src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -39,104 +41,101 @@ public void Dispose()
         {
             _senderCts.Cancel();
             _pollCts.Cancel();
-            _pipeline?.Dispose();
         }
 
-        public Task StartAsync(Uri url, IPipelineConnection pipeline)
+        public Task StartAsync(Uri url, IChannelConnection<Message> toFromConnection)
         {
-            _pipeline = pipeline;
-
-            // Schedule shutdown of the poller when the output is closed
-            pipeline.Output.Writing.ContinueWith(_ =>
-            {
-                _pollCts.Cancel();
-                return TaskCache.CompletedTask;
-            });
+            _toFromConnection = toFromConnection;
 
             // Start sending and polling
-            _poller = Poll(Utils.AppendPath(url, "poll"), _pollCts.Token);
-            _sender = SendMessages(Utils.AppendPath(url, "send"), _senderCts.Token);
+            _poller = Poll(Utils.AppendPath(url, "poll"), _pollCts.Token).ContinueWith(_ => _senderCts.Cancel());
+            _sender = SendMessages(Utils.AppendPath(url, "send"), _senderCts.Token).ContinueWith(_ => _pollCts.Cancel());
             Running = Task.WhenAll(_sender, _poller);
 
             return TaskCache.CompletedTask;
         }
 
-        private async Task Poll(Uri pollUrl, CancellationToken cancellationToken)
+        private async Task Poll(Uri pollUrl, CancellationToken pollCancellationToken)
         {
             try
             {
-                while (!cancellationToken.IsCancellationRequested)
+                while (!pollCancellationToken.IsCancellationRequested)
                 {
                     var request = new HttpRequestMessage(HttpMethod.Get, pollUrl);
                     request.Headers.UserAgent.Add(DefaultUserAgentHeader);
 
-                    var response = await _httpClient.SendAsync(request, cancellationToken);
+                    var response = await _httpClient.SendAsync(request, pollCancellationToken);
                     response.EnsureSuccessStatusCode();
 
-                    if (response.StatusCode == HttpStatusCode.NoContent || cancellationToken.IsCancellationRequested)
+                    if (response.StatusCode == HttpStatusCode.NoContent || pollCancellationToken.IsCancellationRequested)
                     {
                         // Transport closed or polling stopped, we're done
                         break;
                     }
                     else
                     {
-                        // Write the data to the output
-                        var buffer = _pipeline.Output.Alloc();
-                        var stream = new WriteableBufferStream(buffer);
-                        await response.Content.CopyToAsync(stream);
-                        await buffer.FlushAsync();
+                        var ms = new MemoryStream();
+                        await response.Content.CopyToAsync(ms);
+                        var message = new Message(ReadableBuffer.Create(ms.ToArray()).Preserve(), Format.Text);
```

Comment:
No. We are sending it and the receiver is responsible for disposing it. Otherwise we could dispose something they are using.